src/dotnet/MLSearch.Service/Module/MLSearchServiceModule.cs

```diff
@@ -63,9 +63,10 @@ protected override void InjectServices(IServiceCollection services)
         var connectionSettings = new ConnectionSettings(
             new SingleNodeConnectionPool(new Uri(openSearchClusterUri)),
             sourceSerializer: (builtin, settings) => new OpenSearchJsonSerializer(builtin, settings));
+        services.AddSingleton<IndexNames>();
 
         services.AddSingleton<IOpenSearchClient>(_ => new OpenSearchClient(connectionSettings));
-        services.AddSingleton(e => ActivatorUtilities.CreateInstance<ClusterSetup>(e, modelGroupName))
+        services.AddSingleton(e => ActivatorUtilities.CreateInstance<ClusterSetup>(e, HostInfo.IsDevelopmentInstance, modelGroupName))
```

Comment:
I don't want a HostInfo dependency because neither OpenSearchConfigurator nor ClusterSetup is supposed to be hosted, but modules are. However, you are right; sending just one property of HostInfo as a configuration parameter just masks the problem. Instead, configurators should receive an exact number_of_replicas setting.

Reply:
`ClusterSetup` now accepts `OpenSearchSettings` and it has `NumberOfReplicas` value.